integration/PetPortalServiceNowExtended/README.md

```diff
@@ -0,0 +1,6 @@
+# Pet Portal Blueprint
+
+This is my Pet Portal includes extra governance in Service Now.  This blueprint also leverages the xlr-xlrelease-plugin to dynamically tag the releases with the Service Now tickets.
+
```

Comment:
You asked me to remove the `docker-compose.yml` file that defined everything that I needed to to run this template.  Should I try to document here everything that is defined in the [demo-self-service](https://github.com/xebialabs-external/demo-self-service)?

Reply:
We will soon create a home for these docker-compose files, but I believe the docker compose file doesn't document what the user needs to do further or anything, it would just be a step that says something like `run this docker-compose file to start up the required infra` what should the user do after that?

Reply:
Also I don't have access to https://github.com/xebialabs-external/demo-self-service and I dont know what it does, this is a public repo and people who might use this also doesnt know what demo-self-service is unless there is something in Readme which points to a public website/repo

Reply:
The reason I had the docker compose in the README was that the repo is private but the docker images that it refers to are public in `xebialabsunsupported`.  This way someone could create an environment that would work with the blueprint.  How and what would you suggest for documentation here.  From what I can tell the other blueprints don't have any such documentation.  The limited instances that I have found involve downloading a docker-compose.yaml file as well.  Should I be including that as part of my template?

Reply:
I never said the docker-compose is not needed, what said was that we should move them to a different repo and link up here. 
Below is one of the minimal Readme we have(and it serves the purpose) I'm asking for something at least like that. We also have https://github.com/xebialabs/blueprints/blob/development/gcp/microservice-ecommerce/xebialabs/README.md.tmpl since that use case demands a detailed ReadMe. But just adding few lines of text on file doesn't serve any purpose. We could at least add
1. What is this about
2. Prerequisites(download this docker-compose setup, run docker-compose up)
3. Further steps(xl apply ..., go to XLR, start release)

I really don't think I'm asking for anything that is not reasonable here

---
## Data Lake on AWS

The blueprint deploys a data lake architecture to AWS using CloudFormation.
XL Deploy does the provisioning and deployment, while XL Release orchestrates everything.
Refer to https://docs.aws.amazon.com/solutions/latest/data-lake-solution

The application is available in the repo [https://github.com/awslabs/aws-data-lake-solution](https://github.com/awslabs/aws-data-lake-solution).


### Prerequisites

If you do not have XL Release and XL Deploy instances running, you can use this Docker Compose setup to spin them up. The setup also contains a preconfigured Jenkins Instance.
Please note that you need to obtain a trial license or bring your own license for XL Release and XL Deploy. Follow the steps below to use the Docker Compose files.

 1. Download the contents of https://github.com/xebialabs/e-commerce-microservice/tree/master/docker-compose/xebialabs into a folder such as `xl-platform`.
 2. Copy the license for XL Release into `xl-platform/xl-release/default-conf/` and the license for XL Deploy into `xl-platform/xl-deploy/default-conf/`.
 3. Run `docker-compose up --build` on the `xl-platform` folder.

To deploy this blueprint with the XebiaLabs DevOps Platform, follow the steps below:

1. Apply the generated YAML configurations using the XL CLI.

    ```
    xl apply -f xebialabs.yaml
    ```

2. Go to XL Release and look for the data-lake-release-pipeline and start a new release from it.

Reply:
Does it make sense then to include the docker-compose.yml file in the README or in the repo.  While the docker images are public the repo that I’m getting the docker-compose.yml file from is not.  I agree that more information should be provided to make it easier to set up a working environment.  I'm just trying to figure out what the best way to do that.

Reply:
For now, I suggest generating the docker-compose file as well on the root(alongside the xebialabs.yaml). I guess since the private repo is also ours we can decide to open source the docker-compose file as it doesn't have anything proprietary in it. Once we setup the dedicated repo we will move the docker compose file there and update the readme to point to that